content/docs/command-reference/init.md

```diff
@@ -32,88 +33,93 @@ that are hidden from the user. This directory is automatically staged with
-
-- Not enough isolation/granularity - commands like `dvc pull`, `dvc checkout`,
-  and others analyze the whole repository to look for `dvc.yaml` or `.dvc` files
-  to download files and directories, to reproduce <abbr>pipelines</abbr>, etc.
-  It can be expensive in the large repositories with a lot of projects.
-
-- Not enough isolation/granularity - commands like `dvc metrics diff`, `dvc dag`
-  and others by default dump all the metrics, all the pipelines, etc.
+- Many DVC commands can explore the whole <abbr>DVC repository</abbr> to find
+  DVC-tracked data and pipelines to work with. This can be undesirable and
+  inefficient for large monorepos.
 
 #### How does it affect DVC commands?
 
-No matter what mode is used, DVC looks for the `.dvc` directory when it starts
-(from the current working directory and up). Location of the found `.dvc`
-directory determines the root of the DVC project. (In case of `--subdir` it
-might happen that Git repository root is located at different path than the DVC
-project root.)
+The <abbr>project</abbr> root is found by DVC by looking for `.dvc/` from the
+current working directory, up. It defines the scope of action for most DVC
+commands (e.g. `dvc repro`, `dvc pull`, `dvc metrics diff`), meaning that only
+`dvc.yaml`, `.dvc` files, etc. inside the project are usable by the commands.
 
-DVC project root defines the scope for most DVC commands. Mostly meaning that
-all `dvc.yaml` and `.dvc` files under the root path are being analyzed.
+With `--subdir`, the project root will be found before the Git root, making sure
+the scope of DVC commands run here is constrained to this project alone, even if
+there are more DVC-related files elsewhere in the repo. Similarly, DVC commands
+run outside this project root will ignore its contents.
 
-If there are multiple DVC sub-projects but they _are not_ nested, e.g.:
+**Simple structures**: multiple `--subdir` projects, but they are not nested,
+e.g.:
 
-```
+```dvc
 .
-├── .git
-|
+├── .git      # plain Git repo
 ├── project-A
-│   └── .dvc
+│   ├── .dvc    # dvc init --subdir
 │   ...
 ├── project-B
-│   └── .dvc
+│   ├── .dvc    # dvc init --no-scm
 │   ...
 ```
 
-DVC considers them a two separate DVC projects. Any DVC command that is being
-run in the `project-A` is not aware about DVC `project-B`. DVC does not consider
-Git repository root an initialized DVC project in this case and commands that
-require DVC project will raise an error.
+DVC considers them separate projects. Any DVC command run in `project-A` is not
+aware of `project-B`. However, commands that involve versioning (like
+`dvc checkout`) can access the commit history from the Git root (`.`), when run
+in `project-A`.
 
-On the other hand, if there _are_ nested DVC projects, e.g.:
+> `.` is not a DVC project in this case, so most DVC commands can't be run
+> there.
 
-```
-project-A
+**Advanced structures**: If there are nested projects, either `--subdir`,
```

Comment:
I still don't quite understand the motivation behind this tbh? This case is not different from what we already had - we just didn't mention Git in some cases only because it does not matter - projects are isolated git or not ... am I missing something?

Reply:
Yes, a quick summary is that projects are always isolated. Maybe (I should simplify some of the explanations this way) but users specifically wonder about nested Git repos and submodules (seen it in Discord a few times). That's the main motivation here.

Reply:
all these became quite complicated and hard to read with `projects either --subdir or --no-scm` - even for me it's very challenging to wrap my mind around it ... I know that all this subdir stuff hurts the brain- thus it's important to keep it simple, try to not mix, explain essence.

So, for submodules - there no difference at all with regular `dvc init` , right? `--subdir` has nothing to do in this case. DVC commands are isolated (since .dvc defines the scope). If we want to mention this then we can mention we should probably mention it in the description itself? Somewhere where we explain how `dvc init` connects dvc with git.

Same goes for nested Git repos - for DVC (like for git) in this case nothing changes and `--subdir` does not play a role.

Reply:
These 2 block examples (one for not nested and one for nested) were already there (see https://dvc.org/doc/command-reference/init#how-does-it-affect-dvc-commands). I was just expanding the nested one to include other possibilities, but I see how that can be confusing...

> for submodules - there no difference at all with regular dvc init , right? --subdir has nothing to do in this case... we can mention we should probably mention it in the description itself

Correct. OK, I changed it this way in d14e960e.